crates/autopilot/src/boundary/events/cow_amm_factory.rs

```diff
@@ -79,14 +90,21 @@ impl EventStoring<contracts::cow_amm_constant_product_factory::Event> for Indexe
         events: Vec<ethcontract::Event<contracts::cow_amm_constant_product_factory::Event>>,
     ) -> anyhow::Result<()> {
         let mut registry = self.registry.write().await;
-        for event in events {
+        for event in &events {
             let block_number = event
                 .meta
+                .as_ref()
                 .ok_or_else(|| anyhow::anyhow!("Event missing meta"))?
                 .block_number;
-            if let contracts::cow_amm_constant_product_factory::Event::Deployed(event) = event.data
+            if let contracts::cow_amm_constant_product_factory::Event::Deployed(ref event) =
+                event.data
             {
-                registry.insert(block_number, event);
+                registry.insert(block_number, event.clone());
+            }
+            if let Some(sender) = &self.sender {
+                if let Err(e) = sender.send(event.clone()) {
```

Comment:
@MartinquaXD I understand, well we could still fix it with the channels tho, if we were to have something like

```rust
pub enum Event<T> {
   Add(<T>),
   Remove(<T>)
}
``` 
as a channel element, but then it starts growing in complexity. So I guess we should not over-complicate things and go straight with a shareable in-memory resource, wdyt?

Reply:
Another option is to write it in the database and have a notificator, as we have for the block events

Reply:
Eventually we would probably write the stuff into the DB to avoid reindexing all the time. But with more and more events getting added to the DB it might make sense to think about some nicer way to manage that in the DB.
For now I would just have the underlying data shared.